packages/runtime/container-runtime/src/scheduleManager.ts

```diff
@@ -265,30 +267,28 @@ class ScheduleManagerCore {
             return;
         }
 
-        // If the client ID changes then we can move the pause point. If it stayed the same then we need to check.
-        // If batchMetadata is not undefined then if it's true we've begun a new batch - if false we've ended
-        // the previous one
-        if (this.currentBatchClientId !== undefined || batchMetadata === false) {
-            if (this.currentBatchClientId !== message.clientId) {
-                // "Batch not closed, yet message from another client!"
-                throw new DataCorruptionError(
-                    "OpBatchIncomplete",
-                    {
-                        runtimeVersion: pkgVersion,
-                        batchClientId: this.currentBatchClientId,
-                        pauseSequenceNumber: this.pauseSequenceNumber,
-                        localBatch: this.currentBatchClientId === this.getClientId(),
-                        localMessage: message.clientId === this.getClientId(),
-                        ...extractSafePropertiesFromMessage(message),
-                    });
-            }
+        // If we got here, the message is part of a batch. Either starting, in progress, or ending.
+
+        // If this is not the start of the batch, error out if the message was sent by a client other than the one that
+        // started the current batch (it should not be possible for ops from other clients to get interleaved with a batch).
+        if (this.currentBatchClientId !== undefined && this.currentBatchClientId !== message.clientId) {
+            throw new DataCorruptionError(
```

Comment:
`batchMetadata === false` indicates the message that finishes a batch. That should only happen if we're already in the middle of a batch which would make `this.currentBatchClientId !== undefined` true, so the outer condition in the previous version of the code seems to have redundant checks.

If we previously received a message with `batchMetadata === false` and were _not_ in the middle of a batch (i.e. `this.currentBatchClientId === undefined`), the DataCorruptionError would be thrown (unless `message.clientId` had been `undefined` too, which shouldn't happen for a message that finishes a batch?). In the new code that wouldn't happen, but again, a close-batch message should not happen if we're not in the middle of a batch already.

Do we maybe want to add an `assert()` for that?